Linting Javascipt

namangoel1 opened the pull request:
Added JSHint, the JavaScript linter. Runs only on main.js

namangoel1 commented:
I'm going to push an update with JSCS and jQuery's styleguide soon. We might need to figure out styleguides, though - jQ might be too strict. [JSCS' overview](http://jscs.info/overview.html) page lists a few presets, would be great if you could choose one @RoonyH. I prefer jQuery's myself, even though it's a little strict, to be honest.

namangoel1 commented:
Completed [my task](http://www.google-melange.com/gci/task/view/google/gci2014/5836395776049152). Unfortunately, however, the test does not show any helpful error message on failing by the native mocha interface. I have included helpful messages through the console, though.

Also configured jscs in a rough way.

RoonyH commented:
I will look at this tomorrow. Shall we keep only jslint for now? Thanks
much for adding it. More restrictions could be overwhelming for newcomers.
On Jan 4, 2015 1:32 AM, "namangoel1" <notifications@github.com> wrote:

> I'm going to push an update with JSCS and jQuery's styleguide soon. We
> might need to figure out styleguides, though - jQ might be too strict. JSCS'
> overview <http://jscs.info/overview.html> page lists a few presets, would
> be great if you could choose one @RoonyH <https://github.com/RoonyH>. I
> prefer jQuery's myself, even though it's a little strict, to be honest.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/fossasia.github.io/pull/205#issuecomment-68607503>
> .
>

RoonyH commented:
Its not mocha. Mocha tests [pass](https://travis-ci.org/fossasia/fossasia.github.io/jobs/45808173#L307).Travis build is failing due to [gulp](https://travis-ci.org/fossasia/fossasia.github.io/jobs/45808173#L652) ' js linting. I will get this now. And try to get some newcomming students involved in fixing stylings. Meanwhile please work on something more challenging to you :)

Thank you very much these are very valuable work!

namangoel1 commented:
JSHint does not have any option to check indentation, unfortunately. We need JSCS for that. I've fixed all JSHint errors in the code, btw. 

As for Mocha, it was a separate issue I was talking about. In [this task](https://github.com/namangoel1/fossasia.github.io/commit/c70cd958251c474a3ea230ebccc735c76fb6fea8) I convert 'imageSize' to a script that works in the browser. It works, but if it is fails the Mocha UI does not give a helpful message. I have configured the console to output them, though. [Here's what I mean](http://i.imgur.com/PKGRInt.png). I already have put a failure message in the assertion, not sure why Mocha doesn't print that.

Travis fails due to JSCS, yes. JSCS is specifically for styling issues and I think it should be kept. Some rules could be trimmed down, though. I have made some configuration changes and fixed some place in the javascript in this commit.

RoonyH commented on gulpfile.js:
```diff
@@ -1,9 +1,33 @@
+  gulp.src('./javascripts/main.js')
+    .pipe( jshint() )
+    .pipe( jshint.reporter('default') )
+    .pipe( jscs({
```
can we remove all jscs checks for now. since the build is failing due to it? just use `.pipe(jscs())` we will add checks later one by one.

RoonyH commented:
>JSHint does not have any option to check indentation, unfortunately. We need JSCS for that. I've fixed all JSHint errors in the code, btw.
Understood. we'll keep it.

>As for Mocha, it was a separate issue I was talking about. In this task I convert 'imageSize' to a script that works in the browser. It works, but if it is fails the Mocha UI does not give a helpful message. I have configured the console to output them, though. Here's what I mean. I already have put a failure message in the assertion, not sure why Mocha doesn't print that.

aah! sorry for confusion. will look now.
